sdk/clientcore/tools/sdk-codegen-tool/README.md

```diff
@@ -0,0 +1,135 @@
+# Codegen Compile time generator Plugin
+
+A Java annotation processor for generating HTTP service implementations based on annotated interfaces.
+
+## Usage
+
+1. Add the plugin dependency:
+   ```xml
+   <dependencies>
+    <dependency>
+        <groupId>io.clientcore.tools</groupId>
+        <artifactId>sdk-codegen-tool</artifactId>
+        <version>1.0.0.beta.1</version> <!-- {x-version-update;io.clientcore.tools:sdk-codegen-tool;external_dependency} -->
+        <scope>provided</scope>
```

Comment:
Why `provided`?

Reply:
To just make sure that the annotation processor dependency will be available during the compilation phase to generate code, but it will not be included in the final packaged application